### PR TITLE
DOCSUP-11416: Implement IDisk interface for HDFS

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #11058

### CODE REVIEW

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 1]
```diff
@@ -821,4 +823,51 @@ S3 disk can be configured as `main` or `cold` storage:
...
+```
+
+Required parameters:
+
+-   `endpoint` — HDFS endpoint URL in `path`. Endpoint URL should contain a bucket and root path to store data. The parameter is used to look up the HDFS configuration for retrieving files.
+
+Optional parameters:
+
+-   `min_bytes_for_seek` — The minimal number of bytes to use seek operation instead of sequential read. Default value: `1 Mb`.
+-   `thread_pool_size` — The number of threads in the Global Thread pool.
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:873]: I do not think it is worth mentioning this second setting. It is more implementation like and mostly for code compatibility with s3 and the usage benefits of this setting are not proved.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 2]
```diff
@@ -821,4 +823,51 @@ S3 disk can be configured as `main` or `cold` storage:
...
+
+    <merge_tree>
+        <min_bytes_for_wide_part>0</min_bytes_for_wide_part>
+    </merge_tree>
+</yandex>
+```
+
+Required parameters:
+
+-   `endpoint` — HDFS endpoint URL in `path`. Endpoint URL should contain a bucket and root path to store data. The parameter is used to look up the HDFS configuration for retrieving files.
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:868]: In hdfs there are no buckets in the same way as in s3. Here should only mention `root path`. 
Also second sentence does not sound correct. This parameter is **not** to look up **configuration**, but to look up the **files** actually as they are stored by those paths.

**Review by kssenii — CHANGES_REQUESTED** [review]

**Comment by sevirov** [comment]
@kssenii Hi! Please, review this PR again.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 3]
```diff
@@ -762,11 +762,13 @@ Configuration markup:
 ```
 
 Required parameters:
--   `endpoint` — S3 endpoint url in `path` or `virtual hosted` [styles](https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html). Endpoint url should contain bucket and root path to store data.
+
+-   `endpoint` — S3 endpoint URL in `path` or `virtual hosted` [styles](https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html). Endpoint URL should contain a root path to store data.
```
Comment (kssenii) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:766]: Previous explanation was more correct. See https://github.com/ClickHouse/ClickHouse/blob/master/src/IO/S3Common.h#L58-L64.

**Review by kssenii — APPROVED** [review]

**Pull request merged**